rudder-web/src/main/webapp/templates-hidden/common-layout.html

```diff
@@ -131,124 +131,127 @@
-                                    <input type="radio" id="filter-all" checked="true" ng-checked="$parent.filter.all.activated || (!$parent.filter.directive.activated && !$parent.filter.group.activated && !$parent.filter.node.activated && !$parent.filter.parameter.activated && !$parent.filter.rule.activated);"/>All
-                                    <span class="badge pull-right" ng-class="{'label-success':$parent.filter.all.nbResults>0}">{{$parent.filter.all.nbResults}}</span>
+                                  <label for="filter-all" class="btn btn-default" ng-click="$parent.checkFilter($event, true,$parent.filter.all.activated,'all');" ng-class="{active : $parent.noFilterActivated($parent.filter)};">
+                                    <input type="radio" id="filter-all" checked="true" ng-checked="$parent.noFilterActivated($parent.filter);"/>All
+                                    <span class="badge pull-right">
+                                      <span ng-show="$parent.checkSearch(searching,searchStr) && $parent.noFilterActivated($parent.filter);">
+                                        {{$parent.getNumResults("all",results)}}
+                                      </span>
+                                      <span ng-hide="$parent.noFilterActivated($parent.filter);">
+                                        -
+                                      </span>
+                                      <span ng-show="searching && searchStr.length > 2 && $parent.noFilterActivated($parent.filter);" class="loading glyphicon glyphicon-refresh"></span>
+                                    </span>
                                   </label>
                                 </div>
-                        <div class="btn-group group-filters" data-toggle="buttons">
-                                  <label for="filter-directive" class="btn btn-default" ng-click="$parent.checkFilter($event, false,$parent.filter.directive.activated,'directive');">
-                                    <input id="filter-directive" type="checkbox" ng-model="$parent.filter.directive.activated"/> Directive
-                                    <span class="badge pull-right" ng-class="{'label-success':$parent.filter.directive.nbResults>0}">{{$parent.filter.directive.nbResults}}</span>
+                                <div class="btn-group group-filters" data-toggle="buttons">
+                                  <label for="filter-node" class="btn btn-default" ng-click="$parent.checkFilter($event, false,$parent.filter.node.activated,'node');">
+                                    <input id="filter-node" type="checkbox" autocomplete="off" ng-model="$parent.filter.node.activated"/> Nodes
+                                    <span class="badge pull-right">
+                                      <span ng-show=" $parent.checkSearch(searching,searchStr) && $parent.checkParentFilter($parent.filter, 'node');">
```

Comment:
And maybe you could gain much more readiness by doing a function checkFilter ($parent.filter,'node')

that calls $scope.checkSearch () && $scope.checkParentFilter(filter,kind)